### ISSUE TITLE
gen_laguerre returns 0 to a negative exponent

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#1895]
Imported from SourceForge on 2024-07-05 10:24:52
Created by **[andreas-hermann](https://sourceforge.net/u/andreas-hermann/)** on 2020-05-05 14:47:52
Original: https://sourceforge.net/p/maxima/bugs/3631

---

gen_laguerre(1, -1, x);
returns
expt: undefined: 0 to a negative exponent
The same happens e.g. for gen_laguerre(2, -2, x);
There is in fact a simple workaround:
subst(-1, a, ratsimp(gen_laguerre(1, a, x)));
returns
-x
But I would find it more convenient if one could simply call gen_laguerre(1, -1, x);

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 10:24:54
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2020-06-03 17:46:09
Original: https://sourceforge.net/p/maxima/bugs/3631/#92eb

---

Thanks for the bug report. I've experimented with a modification to gen_laguerre which just calls subst(ratsimp(...)) as you mentioned and it seems to work OK in the examples I tried. 

However, I'd be more comfortable implementing a special case of the summation, but I haven't been able to derive anything useful. Here's what I've tried so far:
~~~~
(%i2) assume (n > 0) $

(%i3) declare (n, integer) $

(%i4) load (orthopoly) $

(%i5) gen_laguerre (n, -n, x);
                            n               i1
                           ====    (- n)   x
                           \            i1
                  (1 - n)   >     -------------
                         n /      i1! (1 - n)
                           ====              i1
                           i1 = 0
(%o5)             -----------------------------
                               n!
(%i6) makegamma (%);
            n
           ====                           i1
           \               gamma(i1 - n) x
            >     -----------------------------------
           /      gamma(i1 + 1) gamma((- n) + i1 + 1)
           ====
           i1 = 0
(%o6)      ------------------------------------------
                    gamma(- n) gamma(n + 1)
(%i7) %, n = 2;

gamma: gamma(-2) is undefined.
~~~~
My guess is that the replacement of pochhammer by gamma is assuming that the argument is positive but a different identity should be applied since it's negative. I can look into that more carefully.

Do you know an identity for gen_laguerre(n, -n, x) ? Trying some examples, it seems relatively simple.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 10:24:57
Created by **[andreas-hermann](https://sourceforge.net/u/andreas-hermann/)** on 2020-06-04 07:12:47
Original: https://sourceforge.net/p/maxima/bugs/3631/#92eb/8c68

---

Hi Robert,
 
thanks for your answer.
First, let me say that the error does not only occur for gen_laguerre(n, -n, x) but in fact for gen_laguerre(n, alpha, x) for every alpha = -1, -2, ..., -n. For example

```
(%i2) gen_laguerre(2, alpha, x);
                                           2
                                          x                  2 x
       (alpha + 1) (alpha + 2) (----------------------- - --------- + 1)
                                (alpha + 1) (alpha + 2)   alpha + 1
(%o2)  -----------------------------------------------------------------
                                       2
(%i3) %, alpha = -1;
expt: undefined: 0 to a negative exponent.
 -- an error. To debug this try: debugmode(true);
```

In this example, the error can be avoided by applying ratsimp and I think this is a good workaround for general n.


> Do you know an identity for gen_laguerre(n, -n, x) ? Trying some examples, it seems relatively simple.

We have
```
gen_laguerre(n, -n, x) = (-1)^n * x^n / n!
```
This follows e.g. from the formula for L_n^(alpha) in Abramowitz & Stegun p.775
http://people.math.sfu.ca/~cbm/aands/page_775.htm
and can also be seen from
```
(%i1) declare(n, integer);
(%o1)                                done
(%i2) gen_laguerre(n, -n, x);
                                   n               i1
                                  ====    (- n)   x
                                  \            i1
                         (1 - n)   >     -------------
                                n /      i1! (1 - n)
                                  ====              i1
                                  i1 = 0
(%o2)                    -----------------------------
                                      n!
```
since `(1 - n)_n = 0` for all n = 1, 2, 3,...
so that all the summands in (%o2) are 0 except the one with i1 = n where `(1 - n)_n` in front of the sum cancels with `(1 - n)_n` in the denominator.
However, I cannot get Maxima to see that `(1 - n)_n = 0` even if I assume n>0 and declare n integer. It only works if I substitute a value for n:
```
(%i1) declare(n, integer);
(%o1)                                done
(%i2) assume(n>0);
(%o2)                               [n > 0]
(%i3) pochhammer(1-n, n);
(%o3)                              (1 - n)
                                          n
(%i4) %, n=2;
(%o4)                                  0
```
Do you know how one can do this in Maxima without substituting a value for n?

> My guess is that the replacement of pochhammer by gamma is assuming that the argument is positive but a different identity should be applied since it's negative. I can look into that more carefully.

Maxima does the following replacement
```
(%i5) pochhammer(a, n);
(%o5)                                (a)
                                        n
(%i6) makegamma(%);
                                 gamma(n + a)
(%o6)                            ------------
                                   gamma(a)
and also
(%i7) assume(n>0);
(%o7)                               [n > 0]
(%i8) pochhammer(1-n, n);
(%o8)                             (1 - n)
                                          n
(%i9) makegamma(%);
                                      1
(%o9)                           ------------
                                 gamma(1 - n)
```
Now, gamma(z) has poles at z = 0, -1, -2, ... and thus for n positive integer one should interpret (%o9) as 0. However, Maxima does not do this although we have declared n integer and assumed that n is positive. If one could do this, one could get Maxima to see that pochhammer(1-n, n) = 0.

After all, it still seems to me that using ratsimp is a good workaround. What do you think?

Best,
Andreas

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 10:25:01
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2023-03-11 02:15:13
Original: https://sourceforge.net/p/maxima/bugs/3631/#bfdf

---

Andreas, sorry for the late reply. I'm circling back to this topic after some years. I'll try to post an update soon.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 10:25:04
Created by **[andreas-hermann](https://sourceforge.net/u/andreas-hermann/)** on 2023-03-12 17:24:42
Original: https://sourceforge.net/p/maxima/bugs/3631/#bfdf/d369

---

<html><head></head><body><div style="font-family: Verdana;font-size: 12.0px;"><div>Hello Robert, thank you for your reply. I haven&#39;t thought about this topic for some time. But we can talk about it if there are any open questions related to it.</div>

<div>&nbsp;
<div>&nbsp;
<div name="quote" style="margin:10px 5px 5px 10px; padding: 10px 0 10px 10px; border-left:2px solid #C3D9E5; word-wrap: break-word; -webkit-nbsp-mode: space; -webkit-line-break: after-white-space;">
<div style="margin:0 0 10px 0;"><b>Gesendet:</b>&nbsp;Samstag, 11. M&auml;rz 2023 um 03:15 Uhr<br/>
<b>Von:</b>&nbsp;&quot;Robert Dodier&quot; &lt;robert_dodier@users.sourceforge.net&gt;<br/>
<b>An:</b>&nbsp;&quot;[maxima:bugs] &quot; &lt;3631@bugs.maxima.p.re.sourceforge.net&gt;<br/>
<b>Betreff:</b>&nbsp;[maxima:bugs] #3631 gen_laguerre returns 0 to a negative exponent</div>

<div name="quoted-content">
<div class="markdown_content">
<p>Andreas, sorry for the late reply. I&#39;m circling back to this topic after some years. I&#39;ll try to post an update soon.</p>

<hr/>
<p><strong><a class="alink" href="https://sourceforge.net/p/maxima/bugs/3631/" target="_blank">[bugs:#3631]</a> gen_laguerre returns 0 to a negative exponent</strong></p>

<p><strong>Status:</strong> open<br/>
<strong>Group:</strong> None<br/>
<strong>Labels:</strong> orthopoly<br/>
<strong>Created:</strong> Tue May 05, 2020 02:47 PM UTC by Andreas Hermann<br/>
<strong>Last Updated:</strong> Wed Jun 03, 2020 05:46 PM UTC<br/>
<strong>Owner:</strong> nobody</p>

<p>gen_laguerre(1, -1, x);<br/>
returns<br/>
expt: undefined: 0 to a negative exponent<br/>
The same happens e.g. for gen_laguerre(2, -2, x);<br/>
There is in fact a simple workaround:<br/>
subst(-1, a, ratsimp(gen_laguerre(1, a, x)));<br/>
returns<br/>
-x<br/>
But I would find it more convenient if one could simply call gen_laguerre(1, -1, x);</p>

<hr/>
<p>Sent from sourceforge.net because you indicated interest in <a href="https://sourceforge.net/p/maxima/bugs/3631/" target="_blank">https://sourceforge.net/p/maxima/bugs/3631/</a></p>

<p>To unsubscribe from further messages, please visit <a href="https://sourceforge.net/auth/subscriptions/" target="_blank">https://sourceforge.net/auth/subscriptions/</a></p>
</div>

<div>
<div>&nbsp;</div>
</div>
</div>
</div>
</div>
</div></div></body></html>

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 10:25:08
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2023-03-26 18:07:35
Original: https://sourceforge.net/p/maxima/bugs/3631/#1111

---

- **labels**: orthopoly --> orthopoly, gen_laguerre
- **status**: open --> closed

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 10:25:11
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2023-03-26 18:07:35
Original: https://sourceforge.net/p/maxima/bugs/3631/#5ad0

---

Fixed by commit 9c65247. I implemented the substitution of a specific value into the general formula, as suggested. Thanks for this bug report.
